azure-pipelines.yml

```diff
@@ -202,21 +213,33 @@ stages:
         displayName: '.Net UTs'
 
       - task: PublishPipelineArtifact@1
+        condition: always()
         displayName: 'Save coverage files'
         inputs:
           path: 'coverage'
           artifact: DotnetCoverageNet6
 
       - task: PublishPipelineArtifact@1
+        condition: always()
         displayName: 'Save test results files'
         inputs:
           path: 'TestResults'
           artifact: DotnetTestResultsNet6
 
+      - task: PublishTestResults@2
+        condition: always()
+        displayName: 'Publish test results'
+        inputs:
+          testRunner: VSTest
+          testResultsFiles: '*.trx'
+          searchFolder: 'TestResults'
+          testRunTitle: '$(Agent.JobName)'
+
     - job: dotNetAnalysis
       displayName: '.Net Analysis'
       # This job runs the .Net code analysis and uploads to SonarCloud the test results and coverage reports generated
       # in previous jobs.
+      condition: always()
```

Comment:
We should also run the analysis because we will have code coverage and test reports available and can upload it to SC.